4 - AnimalShelter/Code/AnimalShelter.cs

```diff
@@ -77,5 +81,16 @@ public OperationResult GetAnimalFromID(int uniqueAnimalId)
                 return new OperationResult("Multiple ID matches found - Shelter is in error state");
             }
         }
+        protected virtual bool IsSupported(AnimalType type)
+        {
+            var result =
+                (  type == AnimalType.Bird
+                || type == AnimalType.Cat
+                || type == AnimalType.Dog
+                || type == AnimalType.Snake);
+
+            return result;
+        }
+        private List<Animal> _Animals;
```

Comment:
private variables should be at the top of your class. It was very confusing having this down here and took me a while to find it.

Reply:
Oh also, private variables should not have a capital after the _, eg `_animals`.